pkg/pipelines/config/config.go

```diff
@@ -70,67 +82,61 @@ func (m *Manifest) AddService(envName, appName string, svc *Service) error {
 	return nil
 }
 
-// GetCICDEnvironment returns the CICD Environment if one exists.
-func (m *Manifest) GetCICDEnvironment() (*Environment, error) {
-	envs := []*Environment{}
-	for _, env := range m.Environments {
-		if env.IsCICD {
-			envs = append(envs, env)
+// GetPipelineConfig returns the CICD Environment if one exists.
+func (m *Manifest) GetPipelineConfig() (*Pipeline, error) {
+	if m.Config != nil {
+		if m.Config.PipelineConfig != nil {
+			return m.Config.PipelineConfig, nil
 		}
 	}
-	if len(envs) > 1 {
-		return nil, errors.New("found multiple CI/CD environments")
-	}
-	if len(envs) == 0 {
-		return nil, nil
-	}
-	return envs[0], nil
+	return nil, nil
 }
 
-// GetArgoCDEnvironment returns the ArgoCD Environment if one exists.
-func (m *Manifest) GetArgoCDEnvironment() (*Environment, error) {
-	envs := []*Environment{}
-	for _, env := range m.Environments {
-		if env.IsArgoCD {
-			envs = append(envs, env)
+// GetArgoCDConfig returns the ArgoCD Environment if one exists.
+func (m *Manifest) GetArgoCDConfig() *ArgoCD {
+	if m.Config != nil {
+		if m.Config.ArgoCDConfig != nil {
+			return m.Config.ArgoCDConfig
 		}
 	}
-	if len(envs) > 1 {
-		return nil, errors.New("found multiple ArgoCD environments")
-	}
-	if len(envs) == 0 {
-		return nil, errors.New("could not find ArgoCD environment")
-	}
-	return envs[0], nil
+	return nil
+
 }
 
 // Environment is a slice of Apps, these are the named apps in the namespace.
 //
-// The CICD environment will be used to automatically generate CI/CD resources
-// into.
-// The CICD environment should not have any applications defined.
+
 type Environment struct {
 	Name      string         `json:"name,omitempty"`
 	Pipelines *Pipelines     `json:"pipelines,omitempty"`
 	Services  []*Service     `json:"services,omitempty"`
 	Apps      []*Application `json:"apps,omitempty"`
-	// TODO: this should check that there is 0 or 1 CICD environment in the
-	// manfifest.
-	IsCICD   bool `json:"cicd,omitempty"`
-	IsArgoCD bool `json:"argo,omitempty"`
+}
+
+//Config are the Config environments that constitute the argocd and pipeline environment
```

Comment:
This could be tidied up a bit...

Space before the `Config`.

Move the `PipelineConfig` description down to above the relevant struct field.

Add a similar description for the `ArgoCDConfig`